obspy/imaging/source.py

```diff
@@ -0,0 +1,593 @@
+    # name : COMPONENT              : NED sign and index
+    # NED  : NN, EE, DD, NE, ND, ED : [0, 1, 2, 3, 4, 5]
+    # USE  : UU, SS, EE, US, UE, SE : [1, 2, 0, -5, 3, -4]
+    # RTP  : RR, TT, PP, RT, RP, TP : [1, 2, 0, -5, 3, -4]
+    # DSE  : DD, SS, EE, DS, DE, SE : [1, 2, 0, -5, -3, 4]
+    if coordinate_system == 'RTP' or coordinate_system == 'USE':
+        signs = [1, 1, 1, -1, 1, -1]
+        indices = [1, 2, 0, 5, 3, 4]
+        ned_mt = [sign * mt[ind] for sign, ind in zip(signs, indices)]
+        rtp_mt = mt
+    # for the following option, the moment tensor has to be converted to
+    # RTP coordinates as well
+    # elif coordinate_system == 'DSE':
+    #     signs = [1, 1, 1, -1, -1, 1]
+    #     indices = [1, 2, 0, 5, 3, 4]
+    #     ned_mt = [sign * mt[ind] for sign, ind in zip(signs, indices)]
+    # elif coordinate_system == 'NED':
+    #     ned_mt = mt
+    else:
+        msg = 'moment tensor in {:s} coordinates not implemented yet'
+        raise NotImplementedError(msg.format(coordinate_system))
+
+    # matplotlib plotting is triggered when kind is a list of strings
+    if isinstance(kind, (list, tuple)):
+        fig, axes, kind = _setup_figure_and_axes(kind, fig=fig)
+
+        for ax, kind_ in zip(axes, kind):
+            if kind_ is None:
+                continue
+            elif kind_ == 'p_quiver':
+                _plot_radiation_pattern_quiver(ax, ned_mt, type="P")
+            elif kind_ == 'p_sphere':
+                _plot_radiation_pattern_sphere(
+                    ax, ned_mt, type="P",
+                    p_sphere_direction=p_sphere_direction)
+            elif kind_ == 's_quiver':
+                _plot_radiation_pattern_quiver(ax, ned_mt, type="S")
+            elif kind_ == 's_sphere':
+                _plot_radiation_pattern_sphere(ax, ned_mt, type="S")
+            elif kind_ == 'beachball':
+                ax.spines['left'].set_position('center')
+                ax.spines['right'].set_color('none')
+                ax.spines['bottom'].set_position('center')
+                ax.spines['top'].set_color('none')
+                _plot_beachball(ax, rtp_mt)
+
+        fig.tight_layout(pad=0.1)
+        if show:
+            plt.show()
+        return fig
+
+    elif kind == 'mayavi':
+        _plot_radiation_pattern_mayavi(ned_mt)
+
+    elif kind == 'vtk':
+        # this saves two files, one with the vector field and one
+        # with the nodal lines of the beachball
+        fname_rpattern = 'rpattern.vtk'
+        fname_beachlines = 'beachlines.vtk'
+        _write_radiation_pattern_vtk(
+            ned_mt, fname_rpattern=fname_rpattern,
+            fname_beachlines=fname_beachlines)
+
+    else:
+        raise NotImplementedError('{:s} not implemented yet'.format(kind))
+
+
+def _plot_radiation_pattern_sphere(
+        ax3d, ned_mt, type, p_sphere_direction='inwards'):
+    """
+    private function that plots a radiation pattern sphere into
+    ax3d
```

Comment:
Don't know why this word is on its own line. (There should be a blank line before the parameters, though.)